### PR TITLE
docs(maintaining): add pull requests section

### PR DESCRIPTION
**Pull request by jsjoeio**
This PR adds a section about pull requests and associating with milestones to `MAINTAINING.md`.

## Checklist

- [x] updated `CHANGELOG.md`

### CODE REVIEW

File: docs/MAINTAINING.md [hunk 1]
```diff
@@ -62,4 +65,14 @@ It also gives us a way to separate the issue triage from bigger-picture, long-te
...
+
+## Pull Requests
+
+Ideally, every PR should fix an issue. If it doesn't, make sure it's associated with a version milestone.
+
+If a PR does fix an issue, don't add it to the version milestone. Otherwise, the version milestone will have duplicate information: the issue & the PR fixing the issue.
+
+### Merge Strategies
+
+For most things, we recommend "Squash and Merge". If you're updating `lib/vscode`, we suggest using the "Rebase and Merge" strategy. There may be times where "Create a merge commit" makes sense as well. Use your best judgement.
```
Comment (bpmct) [inline docs/MAINTAINING.md:78]: Awesome to see this included.

**Review by bpmct — APPROVED** [review]
fixed a minor typo with the semver URL in this too

File: docs/MAINTAINING.md (hunk 1 above)
Comment (jsjoeio) [reply docs/MAINTAINING.md:78]: Thanks for fixing that typo too!

Comment (jawnsy) [reply docs/MAINTAINING.md:78]: Examples are always nice - for the "Create a merge commit" flow, if you have any. You might also want to mention that if you're unsure, to discuss with the team?

**Review by oxy — APPROVED** [review]
Looks good!

File: docs/MAINTAINING.md (hunk 1 above)
Comment (jsjoeio) [reply docs/MAINTAINING.md:77]: I don't have a good example off the top of my head but if you have one and want to add it, feel free to add in a follow-up PR? I'll add that last part about discussing with the team though! Good idea.

**Pull request merged**